libs/langchain/langchain/embeddings/vertexai.py

```diff
@@ -1,16 +1,39 @@
-from typing import Dict, List
+import re
+import string
+import threading
+from concurrent.futures import ThreadPoolExecutor, wait
+from typing import Any, Dict, List, Literal, Optional, Tuple
 
 from langchain_core.embeddings import Embeddings
 from langchain_core.pydantic_v1 import root_validator
 
+from langchain.llms.base import create_base_retry_decorator
 from langchain.llms.vertexai import _VertexAICommon
 from langchain.utilities.vertexai import raise_vertex_import_error
 
+_MAX_TOKENS_PER_BATCH = 20000
+_MAX_BATCH_SIZE = 250
+_MIN_BATCH_SIZE = 5
+
 
 class VertexAIEmbeddings(_VertexAICommon, Embeddings):
     """Google Cloud VertexAI embedding models."""
 
-    model_name: str = "textembedding-gecko"
+    model_name: str = "textembedding-gecko@latest"
```

Comment:
Have the `@latest` model being promoted to the default on the SDK level?

I'm we should do this without a long warning on LC side, since many people might be using `VertexAIEmbeddings()` with defaults, and using one model to embed a query and another one to create a vector index might be not a good idea, wdyt?

Reply:
Agree. Let me fix that.

Reply:
Although, at some point the default model will be upgraded anyways, so anyone who didn't specify the version (e.g. `textembedding-gecko@001` will get a new version by default.

Reply:
done